### PR TITLE
chore!: remove deprecated constants

### PR DESCRIPTION
**Pull request by mabdinur**
Removes deprecated constants: ``FILTERS_KEY``, ``NUMERIC_TAGS``, ``LOG_SPAN_KEY``

## Checklist
- [ ] Added to the correct milestone.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

**Comment by mabdinur** [comment]
I combined two deprecation removals to avoid running ci twice. I can add each commit to own PR if that's more desirable.

File: ddtrace/tracer.py [hunk 1]
```diff
@@ -851,44 +851,6 @@ def set_tags(self, tags):
         """
         self._tags.update(tags)
 
-    def _restore_from_shutdown(self):
```
[the remark this answers is not in the document]
Comment (brettlangdon) [reply ddtrace/tracer.py:854]: I mean, it is deprecated, but do we still need to do something on shutdown to `self.start_span`/`self.trace` ?

[the remark this answers is not in the document]
Comment (majorgreys) [reply ddtrace/tracer.py:854]: I would split these changes into two PRs @mabdinur. These will get included in rc2 so no need to rush them both. Thanks for jumping on this!

File: ddtrace/constants.py [hunk 2]
```diff
@@ -39,27 +38,7 @@
...
-    "FILTERS_KEY": "FILTERS",
-    # TODO: Moved to other modules but cannot reference here due to circular imports
-    # ddtrace.contrib.logging.patch._LOG_SPAN_KEY
-    # ddtrace.span._NUMERIC_TAGS
-    "NUMERIC_TAGS": (ANALYTICS_SAMPLE_RATE_KEY,),
-    "LOG_SPAN_KEY": "__datadog_log_span",
-}
-
-
 def __getattr__(name):
```
Comment (mabdinur) [inline ddtrace/constants.py:41]: Should we remove this function or at least deprecate it?

Comment (brettlangdon) [reply ddtrace/constants.py:41]: yeah, I think you can get rid of this and `ensure_pep562` ?

we might need to look for `ensure_pep562` elsewhere too

**Review by brettlangdon — COMMENTED** [review]
you have a good point about removing `__getattr__`, then lgtm

File: ddtrace/constants.py [hunk 3]
```diff
@@ -38,32 +37,4 @@
...
-        )
-        return _DEPRECATED[name]
-
-    if name in globals():
-        return globals()[name]
-
-    raise AttributeError("%s has no attribute %s", __name__, name)
-
-
 ensure_pep562(__name__)
```
Comment (brettlangdon) [inline ddtrace/constants.py:40]: This needs to go away too. This just makes sure `__getattr__` works as intended

Comment (mabdinur) [reply ddtrace/constants.py]: Good catch. I was able to remove the pep562 dependency :)

File: ddtrace/internal/compat.py [hunk 4]
```diff
@@ -232,22 +232,6 @@ def get_connection_response(
     CONTEXTVARS_IS_AVAILABLE = True
 
 
-try:
```
Comment (mabdinur) [inline ddtrace/internal/compat.py:235]: This was only used to support deprecated constants

**Review by majorgreys — APPROVED** [review]

**Review by brettlangdon — APPROVED** [review]

**Comment by mabdinur** [comment]
@mergify backport 1.0.0

**Comment by mabdinur** [comment]
@Mergify backport 1.0

**Pull request merged**